core/src/labels/labels.cpp

```diff
@@ -221,60 +208,121 @@ void Labels::update(const View& _view, float _dt,
 
     /// Collect and update labels from visible tiles
 
-    m_needUpdate = updateLabels(_styles, _tiles, _dt, dz, _view);
+    updateLabels(_styles, _tiles, _dt, dz, _view);
+
+
+    /// Mark labels to skip transitions
+
+    if (int(m_lastZoom) != int(_view.getZoom())) {
```

Comment:
I feel that doing the reverse and slight tweaking to `evalState` might give better result with `skip transition`.